includes/base/atomic-widget-base.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Elementor;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+abstract class Atomic_Widget_Base extends Widget_Base {
+	protected $version = '0.0';
+
+	public function __construct( $data = [], $args = null ) {
+		parent::__construct( $data, $args );
+
+		$this->version = $data['version'] ?? '0.0';
+	}
+
+	abstract public function get_atomic_controls(): array;
+
+	final public function get_controls( $aaa = null ) {
+		return [];
+	}
```

Comment:
I believe we wanted it to return null if a specific `$control_id` is passed (to avoid faulty checks we saw throughout the code that might cause unexpected outcomes)
```suggestion
	final public function get_controls( $control_id = null ) {
		if(!empty($control_id)) {
			return null;
		}
		return [];
	}
```

Reply:
Done.